### PR TITLE
[IMP] payment_stripe: manage async notifications

### PR DESCRIPTION
**Pull request by Demesmaeker**
Some payments may be confirmed asynchronously, and we need to be aware
of their status, even if they are not in a "checkout session complete"
event.

task-2733725

see also:
- https://github.com/odoo/upgrade/pull/3350
- https://github.com/odoo/documentation/pull/1605

### CODE REVIEW

File: addons/payment_stripe/tests/common.py [hunk 1]
```diff
@@ -18,58 +18,20 @@ def setUpClass(cls, chart_template_ref=None):
...
+                    "status": "succeeded",
+                }
             },
-            'id': 'evt_00000000000000',
-            'livemode': False,
-            'object': 'event',
-            'pending_webhooks': 1,
-            'request': None,
-            'type': 'checkout.session.completed',
+            "type": "payment_intent.succeeded"
```
Comment (Feyensv) [inline addons/payment_stripe/tests/common.py:36]: Please add a test to cover the 'setup_intent.succeeded' case :pray:

File: addons/payment_stripe/tests/test_stripe.py [hunk 2]
```diff
@@ -35,16 +35,30 @@ def test_webhook_notification_confirms_transaction(self):
...
         ), patch(
             'odoo.addons.payment_stripe.models.payment_acquirer.PaymentAcquirer'
             '._stripe_make_request',
-            return_value={'status': 'succeeded'},
+            return_value=self.setup_intent_response,
         ):
-            self._make_json_request(url, data=self.notification_data)
-        self.assertEqual(tx.state, 'done')
+            self._make_json_request(
+                url, data=dict(self.notification_data, type="setup_intent.succeeded")
```
Comment (Feyensv) [inline addons/payment_stripe/tests/test_stripe.py:59]: The expected payload is the same for `setup_intent.succeeded` and `payment_intent.succeeded` ?
I was expecting something quite different, but if only the type is expected to differ, then it's ok :eyes:

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Comment by AntoineVDV** [comment]
@robodoo r+

**Pull request closed without merging**